tools/harness-thci/ARM.py

```diff
@@ -1580,25 +1579,102 @@ def startNativeCommissioner(self, strPSKc='GRLpassWord'):
         pass
 
     def startCollapsedCommissioner(self):
-        pass
+        """start openthread stack"""
+        print '%s call startCollapsedCommissioner' % self.port
+        self.__startOpenThread()
+        return 1
 
     def setJoinKey(self, strPSKc):
         pass
 
     def scanJoiner(self, xEUI='*', strPSKd='threadjpaketest'):
-        pass
+        """start commissioner"""
+        print '%s call scanJoiner' % self.port
+        cmd = 'commissioner start %s %s' % (strPSKd, self.provisioningUrl)
+        print cmd
+        self.__sendCommand(cmd)
+        return 1
 
     def setProvisioningUrl(self, strURL='grl.com'):
+        self.provisioningUrl = strURL;
         pass
 
     def allowCommission(self, strPSKc="GRLPassword"):
         pass
 
-    def joinCommissioned(self, strPSKd='GRLpassWordx', waitTime=20):
-        pass
+    def joinCommissioned(self, strPSKd='threadjpaketest', waitTime=20):
+        """start joiner"""
+        print '%s call joinCommissioned' % self.port
+        self.__sendCommand('ifconfig up')
+        self.__sendCommand('panid 0x6666')
+        cmd = 'joiner start %s %s' %(strPSKd, self.provisioningUrl)
+        print cmd
+        self.__sendCommand(cmd)
+        time.sleep(90)
+        self.__sendCommand('thread start')
+        time.sleep(30)
+
+#    def getCommissioningLogs(self):
```

Comment:
Thanks, while I can go through this code to see how it is parsing, it might be simpler if you could post an example log output.